transcode.c

```diff
@@ -2719,6 +2721,14 @@ str_transcode0(int argc, VALUE *argv, VALUE *self, int ecflags, VALUE ecopts)
         }
     }
     else {
+        if (senc && denc && !rb_enc_asciicompat(senc) && !rb_enc_asciicompat(denc)) {
+            VALUE encode_args[2];
+            encode_args[1] = rb_enc_from_encoding(senc);
+            senc = rb_utf8_encoding();
+            encode_args[0] = rb_enc_from_encoding(senc);
+            str = rb_funcallv_kw(str, id_encode, 2, encode_args, RB_NO_KEYWORDS);
```

Comment:
Fixed a bug in `rb_str_conv_enc_opts` at 391abc543cea118a9cd7d6310acadbfa352668ef.

```suggestion
            rb_encoding *utf8 = rb_utf8_encoding();
            str = rb_str_conv_enc(str, senc, utf8);
            senc = utf8;
```